hotel_reservation/models/hotel_reservation.py

```diff
@@ -0,0 +1,1203 @@
+
+    reservation_no = fields.Char("Reservation No", readonly=True)
+    date_order = fields.Datetime(
+        "Date Ordered",
+        readonly=True,
+        required=True,
+        index=True,
+        default=(lambda *a: time.strftime(dt)),
+    )
+    warehouse_id = fields.Many2one(
+        "stock.warehouse",
+        "Hotel",
+        readonly=True,
+        index=True,
+        required=True,
+        default=1,
+        states={"draft": [("readonly", False)]},
+    )
+    partner_id = fields.Many2one(
+        "res.partner",
+        "Guest Name",
+        readonly=True,
+        index=True,
+        required=True,
+        states={"draft": [("readonly", False)]},
+    )
+    pricelist_id = fields.Many2one(
+        "product.pricelist",
+        "Scheme",
+        required=True,
+        readonly=True,
+        states={"draft": [("readonly", False)]},
+        help="Pricelist for current reservation.",
+    )
+    partner_invoice_id = fields.Many2one(
+        "res.partner",
+        "Invoice Address",
+        readonly=True,
+        states={"draft": [("readonly", False)]},
+        help="Invoice address for " "current reservation.",
+    )
+    partner_order_id = fields.Many2one(
+        "res.partner",
+        "Ordering Contact",
+        readonly=True,
+        states={"draft": [("readonly", False)]},
+        help="The name and address of the "
+        "contact that requested the order "
+        "or quotation.",
+    )
+    partner_shipping_id = fields.Many2one(
+        "res.partner",
+        "Delivery Address",
+        readonly=True,
+        states={"draft": [("readonly", False)]},
+        help="Delivery address" "for current reservation. ",
+    )
+    checkin = fields.Datetime(
+        "Expected-Date-Arrival",
+        required=True,
+        readonly=True,
+        states={"draft": [("readonly", False)]},
+    )
+    checkout = fields.Datetime(
+        "Expected-Date-Departure",
+        required=True,
+        readonly=True,
+        states={"draft": [("readonly", False)]},
+    )
+    adults = fields.Integer(
+        "Adults",
+        readonly=True,
+        states={"draft": [("readonly", False)]},
+        help="List of adults there in guest list. ",
+    )
+    children = fields.Integer(
+        "Children",
+        readonly=True,
+        states={"draft": [("readonly", False)]},
+        help="Number of children there in guest list.",
+    )
+    reservation_line = fields.One2many(
+        "hotel_reservation.line",
+        "line_id",
+        "Reservation Line",
+        help="Hotel room reservation details.",
+        readonly=True,
+        states={"draft": [("readonly", False)]},
+    )
+    state = fields.Selection(
+        [
+            ("draft", "Draft"),
+            ("confirm", "Confirm"),
+            ("cancel", "Cancel"),
+            ("done", "Done"),
+        ],
+        "State",
+        readonly=True,
+        default=lambda *a: "draft",
```

Comment:
No need to use the lambda function here.
Simply you can pass as default='draft'

Reply:
Done.